### PR TITLE
Extend all env data

### PR DESCRIPTION
**Pull request by sdtaylor**
I updated `get_env_data()` to get either past data (1981-2013) or future data(2014-2050). Past is the default so everything works the same with other data functions, using `get_env_data(timeframe='future')` now returns data for 2014-2050.

### CODE REVIEW

File: R/forecast-bbs-core.R [hunk 1]
```diff
@@ -229,12 +228,39 @@ get_env_data <- function(){
...
-  env_data <- full_join(bioclim_data, ndvi_data, by = c('site_id', 'year'))
-  env_data <- full_join(env_data, elev_data, by = c('site_id'))
+    
+  ndvi_data = inner_join(ndvi_data_summer, ndvi_data_winter, by=c('site_id','year')) %>%
+    inner_join(ndvi_data_ann, by=c('site_id','year'))
+  
+  #If projecting forward, use NDVI averages for future NDVI values, and use
+  #CMIP5 data for bioclim values
+  if(timeframe == 'future') {
+    ndvi_long_term_averges = ndvi_data %>%
```
Comment (ethanwhite) [inline R/forecast-bbs-core.R]: `averges` -> `averages`

File: R/forecast-bbs-core.R [hunk 2]
```diff
@@ -229,12 +228,39 @@ get_env_data <- function(){
...
+      filter(year %in% 2000:2013) %>%
+      gather(season,value, -site_id, -year) %>%
+      group_by(site_id, season) %>%
+      summarise(value = mean(value)) %>%
+      ungroup()
+    
+    all_sites = unique(ndvi_data$site_id)
+    
+    ndvi_data = expand.grid(site_id = all_sites, year=2014:2050) %>%
+      left_join(ndvi_long_term_averges, by='site_id') %>%
```
Comment (ethanwhite) [inline R/forecast-bbs-core.R]: `averges` -> `averages`

**Comment by davharris** [comment]
Looks great to me. Could you change the dates (e.g. 2013) that are currently hardcoded into adjustable parameters?

**Comment by sdtaylor** [comment]
Sure. So for the NDVI averaging and the years `future` data received?

**Comment by sdtaylor** [comment]
I imagine something like this.

```
ndvi_forecast_source: Years to use for computing the site specific means for an NDVI "forecast". Ignored if timeframe is "past".  
future_data_bounds: years to return if getting future forecast data. Ignored if timeframe is "past"  
get_env_data = function(timeframe='past', ndvi_forecast_source=2000:2013, future_data_bounds=2014:2050){
...
}

**Comment by davharris** [comment]
Looks good to me, except I wouldn't call it "bounds" if it takes a full sequence of years (as opposed to just the endpoints).

**Comment by sdtaylor** [comment]
Hows that?

**Comment by davharris** [comment]
LTGM :-)

**Pull request merged**